common/errors/error.go

```diff
@@ -32,3 +32,22 @@ func (s *ServerError) Model() models.ServerError {
 		Message:   s.Err.Error(),
 	}
 }
+
+func ResolveErrorByEntityType(entityType string) (ErrorCode, error) {
```

Comment:
[nit] this makes the ux for the caller weird, I think it should be handled similar to how errors in `fmt.Printf` are handled.
Also, given we know the caller's scenario, I think it should be something like:
```
func NewEntityDoesNotExistError(entityType, entityID string) *ServerError {
    switch entityType {
    case: "deploy": 
        return Newf(DeployDoesNotExist, "Deploy '%s' does not exist", entityID)
    ...
    default:
        Newf(UnexpectedError, "Entity (type='%s') '%s' does not exist", entityType, entityID) 
    }
}
```

That way, the caller can be like:
```
if len(tags) == 0 {
    return "", "", errors.NewEntityDoesNotExistError(entityType, entityID)
}
```

Reply:
Yeah, that's better

✅